packages/chatbot-server-mongodb-public/src/middleware/useSegmentIds.ts

```diff
@@ -0,0 +1,18 @@
+import { ConversationsMiddleware } from "mongodb-chatbot-server";
+
+export const SEGMENT_USER_ID_HEADER = "X-Segment-User-Id";
+export const SEGMENT_ANONYMOUS_ID_HEADER = "X-Segment-Anonymous-Id";
+
+export function useSegmentIds(): ConversationsMiddleware {
+  return (req, res, next) => {
+    const userId = req.header(SEGMENT_USER_ID_HEADER);
+    const anonymousId = req.header(SEGMENT_ANONYMOUS_ID_HEADER);
+    if (userId) {
+      res.locals.customData.segmentUserId = userId;
+    }
+    if (anonymousId) {
+      res.locals.customData.segmentAnonymousId = anonymousId;
```

Comment:
by including this, does it mean that we're now tracking the segment user and anon IDs in the DB?

Reply:
Yes we will track these in user message `customData`. I think they classify as PII so don't want to keep them in `scrubbed_messages`.